src/QirRuntime/lib/QIR/callables.cpp

```diff
@@ -107,6 +96,60 @@ extern "C"
     }
 }
 
+/*==============================================================================
+    Implementation of QirTupleHeader
+==============================================================================*/
+int QirTupleHeader::AddRef()
+{
+    assert(refCount > 0);
+    return ++refCount;
+}
+
+int QirTupleHeader::Release()
+{
+    --refCount;
+    if (refCount == 0)
+    {
+        char* buffer = reinterpret_cast<char*>(this);
+        delete[] buffer;
+    }
+    return refCount;
+}
+
+QirTupleHeader* QirTupleHeader::Create(int size)
+{
+    assert(size >= 0);

```

Comment:
What does size==0 mean?

Reply:
see the other comment about zero tuple size